psi4/driver/driver_findif.py

```diff
@@ -0,0 +1,214 @@
+        for i in range(Nirrep):
+            if i != freq_irrep_only:
+                salc_indices_pi[i].clear()
+
+    # Define the number of displacements per irrep.
+    Ndisp_pi = []
+
+    for irrep, indices in enumerate(salc_indices_pi):
+        Ndisp = len(indices) * len(disps["asym" if irrep != 0 else "sym"])
+        if mode == "2_0":
+            # Either len(indices) or len(indices)-1 is even, so dividing by two is safe.
+            Ndisp += len(indices) * (len(indices) - 1) // 2 * len(disps["off"])
+        Ndisp_pi.append(Ndisp)
+
+    # Let's print out the number of geometries, the displacement multiplicity, and the CdSALCs!
+    if print_lvl:
+        core.print_out("\tNumber of geometries (including reference) is {:d}.\n".format(sum(Ndisp_pi) + 1))
+        if method_allowed_irreps != 0x1:
+            core.print_out("\tNumber of displacements per irrep:\n")
+            for i, ndisp in enumerate(Ndisp_pi, start=1):
+                core.print_out("\t  Irrep {:d}: {:d}\n".format(i, ndisp))
+
+    if print_lvl > 1:
+        for salc in salc_list:
+            salc.print()
+
+    # We can finally start generating displacements.
+    ref_geom_temp = mol.geometry()
+    ref_geom = ref_geom_temp.clone()
+    ref_geom.name = "Reference geometry"
+
+    disp_geoms = []
+
+    def append_geoms(indices, steps):
+        """Given a list of indices and a list of steps to displace each, append the corresponding geometry to the list."""
+        new_geom = ref_geom.clone()
+        # Next, to make this salc/magnitude composite.
+        index_steps = zip(indices, steps)
+        displace_cart(mol, new_geom, salc_list, index_steps, disp_size)
+        disp_geoms.append(new_geom)
+
+    for h in range(Nirrep):
+        active_indices = salc_indices_pi[h]
+
+        for index in active_indices:
+            # Displace along the diagonal.
+            # Remember that the totally symmetric irrep has special displacements.
+            for val in disps["sym" if h == 0 else "asym"]:
+                append_geoms((index,), val)
+
+        # Hessian from energies? We have off-diagonal displacements to worry about.
+        if mode == "2_0":
+            # i indexes SALC indices of the current irrep.
+            for i, index in enumerate(active_indices):
+                for index2 in active_indices[:i]:
+                    for val in disps["off"]:
+                        append_geoms((index, index2), val)
+
+    disp_geoms.append(ref_geom)
+
+    if print_lvl > 2:
+        for disp in disp_geoms:
+            disp.print_out()
+
+    if print_lvl > 1:
+        core.print_out("\n-------------------------------------------------------------\n")
+
+    return disp_geoms
+
+
+def fd_geoms_1_0(molecule):
+    """Generate geometries for gradient by finite difference of energies.
+       Only symmetric displacements are necessary, so user specification of symmetry is disabled."""
+    return geom_generator(molecule, -1, "1_0")
```

Comment:
Any good things to test besides that existing findif works? maybe number of generated displacements for different irreps, stencils, RT-inclusiveness, user-symmetry of an ethene?